pipelined/bevy_core_pipeline/src/main_pass_3d.rs

```diff
@@ -48,52 +50,155 @@ impl Node for MainPass3dNode {
         world: &World,
     ) -> Result<(), NodeRunError> {
         let view_entity = graph.get_input_entity(Self::IN_VIEW)?;
-        let (transparent_phase, target, depth) = self
+        let (opaque_phase, alpha_mask_phase, transparent_phase, target, depth) = self
             .query
             .get_manual(world, view_entity)
             .expect("view entity should exist");
         let clear_color = world.get_resource::<ClearColor>().unwrap();
-        let pass_descriptor = RenderPassDescriptor {
-            label: Some("main_pass_3d"),
-            color_attachments: &[RenderPassColorAttachment {
-                view: if let Some(sampled_target) = &target.sampled_target {
-                    sampled_target
-                } else {
-                    &target.view
-                },
-                resolve_target: if target.sampled_target.is_some() {
-                    Some(&target.view)
-                } else {
-                    None
-                },
-                ops: Operations {
-                    load: LoadOp::Clear(clear_color.0.into()),
-                    store: true,
-                },
-            }],
-            depth_stencil_attachment: Some(RenderPassDepthStencilAttachment {
-                view: &depth.view,
-                depth_ops: Some(Operations {
-                    load: LoadOp::Clear(0.0),
-                    store: true,
+
+        {
```

Comment:
I think this is fine for now, but this seems ripe for parallel command encoding (once we start doing that). Breaking this up into 3 separate nodes might make that easier.